### PR TITLE
refactor(compactor): delete all at once algorithm

### PR DESCRIPTION
**Pull request by alamb**
# Rationale
The compactor is going to be complicated no matter what we do, as it is doing something complicated. So let's keep the code as simple as possible (like not parameterized by algorithm as well)

Now that the new algorithm has been working for several days,  I think we can be confident enough that we don't need to revert to the old algorithm

### CODE REVIEW

File: compactor2/tests/integration.rs [hunk 1]
```diff
@@ -163,78 +108,23 @@ async fn test_compact_all_at_once() {
...
+            (1, CompactionLevel::FileNonOverlapped),
+            (2, CompactionLevel::Initial),
+            (3, CompactionLevel::Initial),
+            (4, CompactionLevel::FileNonOverlapped),
+            (5, CompactionLevel::Initial),
+            (6, CompactionLevel::Initial),
         ],
-        &batches
     );
 }
```
Comment (alamb) [inline compactor2/tests/integration.rs:129]: Github renders this diff in a confusing manner by default in my opinion -- I found https://github.com/influxdata/influxdb_iox/pull/7004/files?w=1 easier to see what changed

Comment (domodwyer) [reply compactor2/tests/integration.rs:129]: FYI this can be (and for me, is) set as the default for diffs.

**Review by domodwyer — APPROVED** [review]

**Comment by alamb** [comment]
As a not to my (future) self at @domodwyer 's excellent suggestion I also double checked that this algorithm is no longer used on our production clusters: https://influxdata.slack.com/archives/C04MRQ772P8/p1676637255522829?thread_ts=1676634222.007779&cid=C04MRQ772P8

File: compactor2/src/components/file_filter/and.rs [hunk 2]
```diff
@@ -10,6 +10,7 @@ pub struct AndFileFilter {
 }
 
 impl AndFileFilter {
+    #[allow(dead_code)]
```
Comment (NGA-TRAN) [inline compactor2/src/components/file_filter/and.rs:13]: Oh, we no longer use the AND operator? I agree we might be able to reuse it soon and let us keep it

File: compactor2/src/components/hardcoded.rs [hunk 3]
```diff
@@ -134,7 +134,7 @@ pub fn hardcoded_components(config: &Config) -> Arc<Components> {
     partition_filters.push(Arc::new(MaxNumColumnsPartitionFilter::new(
         config.max_num_columns_per_table,
     )));
-    partition_filters.append(&mut version_specific_partition_filters(config));
+    partition_filters.append(&mut make_partition_filters(config));
```
Comment (NGA-TRAN) [inline compactor2/src/components/hardcoded.rs:137]: ❤️

File: compactor2/src/components/hardcoded.rs [hunk 4]
```diff
@@ -275,7 +275,7 @@ pub fn hardcoded_components(config: &Config) -> Arc<Components> {
         round_info_source: Arc::new(LoggingRoundInfoWrapper::new(Arc::new(
             LevelBasedRoundInfo::new(config.max_num_files_per_plan),
         ))),
-        files_filter: version_specific_files_filter(config),
+        files_filter: Arc::new(FilesFilterChain::new(vec![])),
```
Comment (NGA-TRAN) [inline compactor2/src/components/hardcoded.rs:278]: This `files_filter` is used for AllAtOnce only, it is safe to remove this too.
And when you remove this, you will see you need to remove this https://github.com/influxdata/influxdb_iox/blob/bda2310ca1a5ac385a88f623a8bbe7f4c9b96f39/compactor2/src/driver.rs#L192, too

File: compactor2/src/config.rs [hunk 5]
```diff
@@ -98,9 +98,6 @@ pub struct Config {
     /// Shard config (if sharding should be enabled).
     pub shard_config: Option<ShardConfig>,
 
-    /// Version of the compaction algorithm.
-    pub compact_version: AlgoVersion,
-
```
Comment (NGA-TRAN) [inline compactor2/src/config.rs:103]: 🎉

**Review by NGA-TRAN — APPROVED** [review]
Looks great. Thanks @alamb 
There are 2 more lines you can remove 🙂

**Comment by alamb** [comment]
Per internal discussion I think we have decided to not merge this right before the weekend, but instead will do at some time when we have full workday capacity (either early next week or the week after)

File: compactor2/src/components/hardcoded.rs (hunk 4 above)
Comment (alamb) [reply compactor2/src/components/hardcoded.rs]: Removed in [d90443d](https://github.com/influxdata/influxdb_iox/pull/7004/commits/d90443d9e63e49a4b5e9fa5339a66ef3c4751cc4)

**Pull request merged**